Sources/X509/Extensions.swift

```diff
@@ -133,48 +137,77 @@ extension Certificate.Extensions: Hashable { }
         get {
             self._extensions[position]
         }
     }
+}
+
+
+// MARK: Modifying methods
+extension Certificate.Extensions {
 
     /// Append a new ``Certificate/Extension`` into this set of ``Certificate/Extensions-swift.struct``.
     ///
     /// - Parameter ext: The ``Certificate/Extension`` to insert.
+    /// - Throws: If an ``Certificate/Extension`` with the same ``Certificate/Extension/oid`` is already present
     @inlinable
-    public mutating func append(_ ext: Certificate.Extension) {
-        // TODO(cory): enforce uniqueness
-        self._extensions.append(ext)
+    public mutating func append(_ extension: Certificate.Extension) throws {
+        if let oldExtension = self._extensions.first(where: { $0.oid == `extension`.oid }) {
+            throw CertificateError.duplicateOID(reason: "tried to append an extension for OID \(`extension`.oid) which is already present. Old extension: \(oldExtension) New extension: \(`extension`)")
+        } else {
+            self._extensions.append(`extension`)
+        }
     }
-
-    /// Append a sequence of new ``Certificate/Extension``s into this set of ``Certificate/Extensions-swift.struct``.
+    
+    /// Updates the ``Certificate/Extension`` stored in the dictionary for the ``Certificate/Extension/oid`` of the `extension`,
+    /// or appends `extension` if an ``Certificate/Extension`` with same  ``Certificate/Extension/oid`` does not exist.
     ///
-    /// - Parameter extensions: The sequence of new ``Certificate/Extension``s to insert.
+    /// - Parameter extension: The ``Certificate/Extension`` to update or append.
+    /// - Returns: The old ``Certificate/Extension`` that was replaced or `nil` if no ``Certificate/Extension`` with same ``Certificate/Extension/oid`` was present
     @inlinable
-    public mutating func append<Extensions: Sequence>(contentsOf extensions: Extensions) where Extensions.Element == Certificate.Extension {
-        // TODO(cory): enforce uniqueness
-        self._extensions.append(contentsOf: extensions)
+    @discardableResult
+    public mutating func update(_ extension: Certificate.Extension) -> Certificate.Extension? {
+        if let index = self._extensions.firstIndex(where: { $0.oid == `extension`.oid }) {
+            let oldExtension = self._extensions[index]
+            self._extensions[index] = `extension`
+            return oldExtension
+        } else {
+            self._extensions.append(`extension`)
+            return nil
+        }
+    }
+    
+    /// Removes the ``Certificate/Extension`` with the given `oid`.
+    /// - Parameter oid: The  ``Certificate/Extension/oid`` of the``Certificate/Extension`` to remove.
+    /// - Returns: The ``Certificate/Extension`` that was removed,
+    ///     or `nil` if an ``Certificate/Extension`` was not present in with the given `oid`.
+    @inlinable
+    @discardableResult
+    public mutating func remove(for oid: ASN1ObjectIdentifier) -> Certificate.Extension? {
```

Comment:
I don't think `for` is the right name for this parameter. Probably fine to have the parameter name be `_`.